plugins/outputs/iotdb/iotdb_test.go

```diff
@@ -0,0 +1,377 @@
+	testClient.TreatTagsAs = "fields"
+
+	result, err := testClient.ConvertMetricsToRecordsWithTags(testMetrics001)
+	require.NoError(t, err)
+	err = testClient.modifyRecordsWithTags(result)
+	require.NoError(t, err)
+	testRecordsWithTags004 := recordsWithTags{
+		DeviceIDList: []string{"root.computer.fan", "root.computer.fan", "root.computer.keyboard"},
+		MeasurementsList: [][]string{
+			{"temperature", "counter", "owner", "price"}, {"temperature", "counter", "owner", "price"},
+			{"temperature", "counter", "unsigned_big", "string", "bool", "int_text"},
+		},
+		ValuesList: [][]interface{}{
+			{float64(42.55), int64(987654321), "cpu", "expensive"},
+			{float64(56.24), int64(123456789), "gpu", "cheap"},
+			{float64(30.33), int64(123456789), int64(math.MaxInt64), "Made in China.", bool(false), "123456789011"},
+		},
+		DataTypesList: [][]client.TSDataType{
+			{client.DOUBLE, client.INT64, client.TEXT, client.TEXT},
+			{client.DOUBLE, client.INT64, client.TEXT, client.TEXT},
+			{client.DOUBLE, client.INT64, client.INT64, client.TEXT, client.BOOLEAN, client.TEXT},
+		},
+		TimestampList: []int64{
+			constTestTimestamp.UnixNano(), constTestTimestamp.UnixNano(), constTestTimestamp.UnixNano(),
+		},
+	}
+	require.True(t, compareRecords(result, &testRecordsWithTags004, testClient.Log))
+	if testing.Short() {
+		t.Skip("Skipping integration test in short mode")
+	}
+	require.NoError(t, testConnectWriteMetricInThisConf(testClient, testMetrics001))
+}
+
+// Test Tags modification in method 'device_id'
+func TestTagsConversion005(t *testing.T) {
+	testClient := newTestClient()
+	testClient.TreatTagsAs = "device_id"
+
+	result, err := testClient.ConvertMetricsToRecordsWithTags(testMetrics001)
+	require.NoError(t, err)
+	err = testClient.modifyRecordsWithTags(result)
+	require.NoError(t, err)
+	testRecordsWithTags005 := recordsWithTags{
+		DeviceIDList: []string{"root.computer.fan.cpu.expensive", "root.computer.fan.gpu.cheap", "root.computer.keyboard"},
+		MeasurementsList: [][]string{
+			{"temperature", "counter"}, {"temperature", "counter"},
+			{"temperature", "counter", "unsigned_big", "string", "bool", "int_text"},
+		},
+		ValuesList: [][]interface{}{
+			{float64(42.55), int64(987654321)},
+			{float64(56.24), int64(123456789)},
+			{float64(30.33), int64(123456789), int64(math.MaxInt64), "Made in China.", bool(false), "123456789011"},
+		},
+		DataTypesList: [][]client.TSDataType{
+			{client.DOUBLE, client.INT64},
+			{client.DOUBLE, client.INT64},
+			{client.DOUBLE, client.INT64, client.INT64, client.TEXT, client.BOOLEAN, client.TEXT},
+		},
+		TimestampList: []int64{
+			constTestTimestamp.UnixNano(), constTestTimestamp.UnixNano(), constTestTimestamp.UnixNano(),
+		},
+	}
+	require.True(t, compareRecords(result, &testRecordsWithTags005, testClient.Log))
+	if testing.Short() {
+		t.Skip("Skipping integration test in short mode")
+	}
+	require.NoError(t, testConnectWriteMetricInThisConf(testClient, testMetrics001))
+}
```

Comment:
Please use a table based test here. That is, define the different test-cases and then run them in a loop instead of recoding the program flow. See e.g. the [exec tests](https://github.com/influxdata/telegraf/blob/3ad9fe73ba6b6b9ab2ccb25fa692aecc738bd461/plugins/outputs/exec/exec_test.go#L17) for an example.

Reply:
All right! It's fixed. All test-cases are in one table know. Code is shorter.